packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2915,6 +2916,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             throw new UsageError("can't get state when offline load disabled");
         }
 
+        const pendingMessages = this.pendingStateManager.getLocalState();
+        const firstMessage = pendingMessages?.pendingStates[0] as IPendingMessage;
+        const isFirstMessageBatchFirst = firstMessage.opMetadata?.batch;
+        const lastMessage = pendingMessages?.pendingStates[pendingMessages.pendingStates.length - 1].type;
+        assert(!!isFirstMessageBatchFirst, "First message must have batch metadata");
```

Comment:
Is this true? What happens if there is a single message in the batch? The batch metadata is removed in that case right?

Reply:
I think the metadata is removed by the schedule manager right before the delta manager actually sends them. This should be ok from the perspective of the pending state manager, as it deals with local batches.

Reply:
I need to add extra case for single message